BaikalDataGenerator/Source/main.cpp

```diff
@@ -30,11 +30,48 @@ void Run(const DGenConfig& config)
 
     Render render(config.scene_file, config.width, config.height);
 
-    render.GenerateDataset(config_loader.CamStatesBegin(), config_loader.CamStatesEnd(),
-                           config_loader.LightsBegin(), config_loader.LightsEnd(),
-                           config_loader.SppBegin(), config_loader.SppEnd(),
-                           config.output_dir,
-                           config.gamma_correction);
+    if (config.split_num != 0)
+    {
+        if (static_cast<size_t>(config.split_num) > config_loader.CamStates().size())
+        {
+            THROW_EX("'split_num' option value is bigger than camera states number");
+        }
+
+        if (config.split_idx >= config.split_num)
```

Comment:
also config.split_ids should be non-negative

Reply:
'config.split_idx' is unsigned